_posts/2022-12-12-deltalake-spark-testing-learnings.md

```diff
@@ -0,0 +1,232 @@
+---
+layout: post
+font: serif
+title: Databricks - An Approach To Unit Testing
+date: 2022-10-27
+highlight: monokai
+author: Bryan Yu
+image: /img/blog/databricks/databricks.png
+tags: databricks deltatable dataframes spark workflows integration unit testing
+description: Lets look at testing our code using delta lake and spark
+---
+
+In the world of software development, unit testing is a concept already understood and quite widely adopted as good practice throughout various teams and industries. You write a failing test, then write the code to make it pass and rinse repeat if you closely follow Test Driven Development. 
+
+On the other hand, in the data land however this is not usually an area explored much at all as engineers would code up some complex SQL statements and transformation tasks in the data software of their choice but not store it in version control nor having automated tests to verify it still works as intended as it goes through iterative changes.
+
+If you are using a tool like Snowflake, you can easily write integration tests for things like your procedures or UDFs if you use them to run your data transformation tasks. With Databricks however, the landscape is a bit different in some aspects in that we can also dip our hands into unit testing, so lets find out why!
+
+Before we commence, please note that any examples and discussions below regarding unit testing will be focused on users who are currently writing their code in `python 3` and running orchestrations via `Databricks Workflows` as `Delta Live Tables` are a separate topic altogether. 
+
+The following topics will be covered here as we make our way to unit testing. If you are already familiar with points 1-3, them skip to the last section at the bottom `Unit Testing`: 
+
+1. Delta Lake
+2. Apache Spark & PySpark
+3. Integration Testing 
+4. Unit Testing
```

Comment:
Suggestion: Add anchors to the headings below and make these clickable links.
```suggestion
1. [Delta Lake](#delta-lake)
2. [Apache Spark & PySpark](#pyspark)
3. [Integration Testing](#integration-testing) 
4. [Unit Testing](#unit-testing)
```

Reply:
oo nifty thanks